linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -856,6 +856,10 @@ private final class ClassDefChecker(classDef: ClassDef,
         transient.traverse(new Traversers.Traverser {
           override def traverse(tree: Tree): Unit = checkTree(tree, env)
         })
+
+      case LinkTimeIf(_, thenp, elsep) =>
+        checkTree(thenp, env)
+        checkTree(elsep, env)
```

Comment:
I wonder if we should "type-check" the `cond` here? Including making sure that all properties are valid. This way we would know that everything afterwards can resolve the conditions.

It's not ideal because it means that, once we add custom link-time properties, ClassDef checking becomes dependent on the `CoreSpec`, which is not currently the case.

Perhaps it would be better to report errors in the `cond` as Analyzer errors. At least Analyzer errors are understood to be user-level errors, whereas ClassDef checking errors should always indicate a compiler bug.

Reply:
Now we "type-check" the `cond` that

- The top-level is typed boolean
- `lhs` and `rhs` has the same type in `BinaryOp`
- `Int_*` has int types for `lhs` and `rhs` / same applies to `Boolean_*` ops